src/altair/helpers.rs

```diff
@@ -68,9 +103,36 @@ pub fn get_next_sync_committee<
         MAX_VALIDATORS_PER_COMMITTEE,
         SYNC_COMMITTEE_SIZE,
     >,
-    _context: &Context,
-) -> Result<()> {
-    Ok(())
+    context: &Context,
+) -> Result<SyncCommittee<SYNC_COMMITTEE_SIZE>> {
+    // Return the next sync committee, with possible pubkey duplicates.
+    let indices = get_next_sync_committee_indices(state, context)?;
+    let public_keys = state
+        .validators
+        .iter()
+        .enumerate()
+        .filter_map(|(i, v)| {
+            if indices.contains(&i) {
+                Some(&v.public_key)
+            } else {
+                None
+            }
+        })
+        .collect::<Vec<_>>();
```

Comment:
try this:

```suggestion
                Some(v.public_key.clone())
            } else {
                None
            }
        })
        .collect::<Vector<_, SYNC_COMMITTEE_SIZE>>();
```

Reply:
you can skip one intermediate allocated `Vec` this way

and it is ok to clone the public keys here but there will just be different sets

you can imagine some kind of optimized implementation that only tracks indices of pubkeys but simpler to just copy them over for now

Reply:
oh also to be clear you can drop the line below that tries to build `public_keys`